### ISSUE TITLE
PhysX_StaticRigidBody_Staging: PhysX Prismatic Joint entity can cause an assert to be printed to Console

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#14389]
**Describe the bug**
An entity that has PhysX Prismatic Joint and PhysX Dynamic Rigid Body components added to it can cause an assert to be printed to Console when entering Game Mode. Please note that:
* It could not have been reproduced on the Development (9291f40) branch.
* On one of the machines on the PhysX_StaticRigidBody_Staging (5d4392e) branch it did not occur at first and after some time testing around this and other Joint components it was reproducing 100% of the time.
* On another machine on the PhysX_StaticRigidBody_Staging (5d4392e) branch this did not reproduce at all at first, using various different suggested reproduction steps, but after deleting Cache it started occuring.

Assert information:
File:
```
C:/o3de/Gems/PhysX/Code/Source/PrismaticJointComponent.cpp
```
Condition:
```
(16456) 'void __cdecl PhysX::PrismaticJointComponent::CachePhysXD6Joint(void)'
 It is not PxD6Joint
```
Line: 113

**Steps to reproduce**
1. Create an entity, add PhysX Dynamic Rigid Body and PhysX Prismatic Joint components to it.
2. Enter the Game Mode.
3. (If no assert occurs) Close the Editor, delete <project-path\>/Cache folder and redo steps 1-2 after Asset Processor reprocesess the assets.

**Expected behavior**
Nothing of note happens.

**Actual behavior**
An assert is printed to the Console.

**Video and Editor.log**

https://user-images.githubusercontent.com/86952082/216630158-abbb09ce-03c7-4576-8564-c25240422859.mp4

[Editor.log](https://github.com/o3de/o3de/files/10579818/Editor.log)

**Found in Branch**
PhysX_StaticRigidBody_Staging (5d4392e)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
5d4392e368d7206b3beddde9c59abdedd7855774

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 11
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by moraaar** [comment]
Fixed in `PhysX_StaticRigidBody_Staging` branch.

**Comment by LB-ArturZieba** [comment]
As the PhysX_StaticRigidBody_Staging branch was merged into the o3de/o3de Development branch I have verified this issue on it instead. Issue verified as fixed on the Development (7ce6550) build.
